src/ipa/rpi/controller/rpi/awb_nn.cpp

```diff
@@ -0,0 +1,505 @@
+	std::string modelPath = "/ipa/rpi/pisp/awb_model.tflite";
+	if (nnConfig_.model.empty()) {
+		std::string root = utils::libcameraSourcePath();
+		if (!root.empty()) {
+			modelPath = root + modelPath;
+		} else {
+			modelPath = LIBCAMERA_DATA_DIR + modelPath;
+		}
+
+		if (!File::exists(modelPath)) {
+			LOG(RPiAwb, Error) << "No model file found in standard locations";
+			nnConfig_.enable_nn = false;
+			return;
+		}
+	} else {
+		modelPath = nnConfig_.model;
+	}
+
+	model_ = tflite::FlatBufferModel::BuildFromFile(modelPath.c_str());
+
+	if (!model_) {
+		LOG(RPiAwb, Error) << "Failed to load model from " << modelPath;
+		nnConfig_.enable_nn = false;
+		return;
+	}
+
+	tflite::MutableOpResolver resolver;
+	tflite::ops::builtin::BuiltinOpResolver builtin_resolver;
+	resolver.AddAll(builtin_resolver);
+	tflite::InterpreterBuilder(*model_, resolver)(&interpreter_);
+	if (!interpreter_) {
+		LOG(RPiAwb, Error) << "Failed to build interpreter for model " << nnConfig_.model;
+		nnConfig_.enable_nn = false;
+		return;
+	}
+
+	interpreter_->AllocateTensors();
+	TfLiteTensor *input_tensor = interpreter_->input_tensor(0);
+	TfLiteTensor *output_tensor = interpreter_->output_tensor(0);
+	if (!input_tensor || !output_tensor) {
+		LOG(RPiAwb, Error) << "Model missing input or output tensor";
+		nnConfig_.enable_nn = false;
+		return;
+	}
+
+	const int expected_input_size = 32 * 32 * 3;
+	int actual_input_size = 1;
+	for (int i = 0; i < input_tensor->dims->size; i++) {
+		actual_input_size *= input_tensor->dims->data[i];
+	}
+
+	if (actual_input_size != expected_input_size) {
+		LOG(RPiAwb, Error) << "Model input tensor size mismatch. Expected: "
+				   << expected_input_size << ", Got: " << actual_input_size;
+		nnConfig_.enable_nn = false;
+		return;
+	}
+
+	int actual_output_size = 1;
+	for (int i = 0; i < output_tensor->dims->size; i++) {
+		actual_output_size *= output_tensor->dims->data[i];
+	}
+
+	if (actual_output_size != nnConfig_.numOutputs) {
+		LOG(RPiAwb, Error) << "Model output tensor size mismatch. Expected: "
+				   << nnConfig_.numOutputs << ", Got: " << actual_output_size;
+		nnConfig_.enable_nn = false;
+		return;
+	}
+
+	if (input_tensor->type != kTfLiteFloat32 || output_tensor->type != kTfLiteFloat32) {
+		LOG(RPiAwb, Error) << "Model input and output tensor must be float32";
+		nnConfig_.enable_nn = false;
+		return;
+	}
+
+	LOG(RPiAwb, Info) << "Model loaded successfully from " << modelPath;
+	LOG(RPiAwb, Debug) << "Model validation successful - Input: " << actual_input_size
+			   << " floats, Output: " << actual_output_size << " floats";
+}
+
+void AwbNN::initialise()
+{
+	Awb::initialise();
+
+	if (nnConfig_.enable_nn) {
+		loadModel();
+		if (!nnConfig_.enable_nn) {
+			LOG(RPiAwb, Warning) << "Neural Network AWB failed to load - switch to Grey method";
+			config_.greyWorld = false;
+			config_.sensitivityR = config_.sensitivityB = 1.0;
+		}
+	}
+}
+
+void AwbNN::transverseSearch(double t, double &r, double &b)
```

Comment:
I sometimes wonder if we should do a very short search along the curve too, but not sure. TBD, I guess.